onnxruntime/core/framework/cloud_invoker.cc

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifdef USE_CLOUD
+#include "http_client.h"
+#include "core/common/common.h"
+#include "core/framework/cloud_invoker.h"
+#include "core/framework/ort_value.h"
+
+#define CHECK_TRITON_ERR(ret, msg)                                                           \
+  if (!ret.IsOk()) {                                                                         \
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, msg, ", triton err: ", ret.Message().c_str()); \
+  }
+
+namespace onnxruntime {
+
+namespace tc = triton::client;
+
+constexpr char* kCloudUri = "cloud.uri";
+constexpr char* kCloudModelName = "cloud.model_name";
+constexpr char* kCloudModelVer = "cloud.model_version";
+constexpr char* kCloudVerbose = "cloud.verbose";
+constexpr char* kCloudEndpointType = "cloud.endpoint_type";
+constexpr char* kCloudAuthKey = "cloud.auth_key";
+constexpr char* kCloudTriton = "triton";
+
+class TritonInvoker : public CloudEndPointInvoker {
+ public:
+  TritonInvoker(const CloudEndPointConfig& config);
+  onnxruntime::Status Send(const CloudEndPointConfig& run_options,
```

Comment:
Why noexcept? Is it guaranteed to not throw any exceptions?

Reply:
removed.